### PR TITLE
Move MakeBoss/MakeNotBoss to 'post_inventory_application'

### PR DESCRIPTION
**Pull request by shadow93**
Also add debug to check for living spectators.

### CODE REVIEW

**Comment by shadow93** [comment]
Oh hey, also managed to fix #284.

Should also fix #277

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 1]
```diff
@@ -4557,6 +4578,35 @@ public Action:event_player_spawn(Handle:event, const String:name[], bool:dontBro
...
+	FF2flags[client]|=FF2FLAG_USEBOSSTIMER;
+	return Plugin_Continue;
+}
+
+/*
+	Calling MakeBoss/MakeNotBoss from 'post_inventory_application' instead of 'player_spawn' 
+	avoids the whole living spectator bug entirely. Also ensures bosses / players gain their 
+	modified weapons back if using TF2_RegeneratePlayer to replenish them. Technically, it's
+	impossible for a living spectator to have a 'post_inventory_application' event fire.
+*/
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: :fire: this whole comment please

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 2]
```diff
@@ -4557,6 +4578,35 @@ public Action:event_player_spawn(Handle:event, const String:name[], bool:dontBro
...
+public Action:Event_PostInventoryApplication(Handle:event, const String:name[], bool:dontBroadcast)
+{
+	if(!Enabled)
+	{
+		return Plugin_Continue;
+	}
+
+	new client=GetClientOfUserId(GetEventInt(event, "userid"));
+	if(!IsValidClient(client))  //I...what.  Apparently this is needed though?
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: You can probably get rid of this comment now :)

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 3]
```diff
@@ -4557,6 +4578,35 @@ public Action:event_player_spawn(Handle:event, const String:name[], bool:dontBro
...
+	return Plugin_Continue;
+}
+
+/*
+	Calling MakeBoss/MakeNotBoss from 'post_inventory_application' instead of 'player_spawn' 
+	avoids the whole living spectator bug entirely. Also ensures bosses / players gain their 
+	modified weapons back if using TF2_RegeneratePlayer to replenish them. Technically, it's
+	impossible for a living spectator to have a 'post_inventory_application' event fire.
+*/
+public Action:Event_PostInventoryApplication(Handle:event, const String:name[], bool:dontBroadcast)
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Nit: Can this be `OnPostInventoryApplication`?

**Comment by shadow93** [comment]
I figured, might as well have all the events start with On followed by the event name.

**Pull request merged**